x-pack/plugins/lens/public/visualizations/metric/dimension_editor.tsx

```diff
@@ -0,0 +1,405 @@
+            ...state,
+            collapseFn,
+          });
+        }}
+      />
+    </>
+  );
+}
+
+function MaximumEditor({ setState, state }: Props) {
+  const idPrefix = htmlIdGenerator()();
+  return (
+    <EuiFormRow
+      label={i18n.translate('xpack.lens.metric.progressDirectionLabel', {
+        defaultMessage: 'Progress bar direction',
+      })}
+      fullWidth
+      display="columnCompressed"
+    >
+      <EuiButtonGroup
+        isFullWidth
+        buttonSize="compressed"
+        legend={i18n.translate('xpack.lens.metric.progressDirectionLabel', {
+          defaultMessage: 'Progress bar direction',
+        })}
+        data-test-subj="lnsMetric_progress_direction_buttons"
+        name="alignment"
+        options={[
+          {
+            id: `${idPrefix}vertical`,
+            label: i18n.translate('xpack.lens.metric.progressDirection.vertical', {
+              defaultMessage: 'Vertical',
+            }),
+            'data-test-subj': 'lnsMetric_progress_bar_vertical',
+          },
+          {
+            id: `${idPrefix}horizontal`,
+            label: i18n.translate('xpack.lens.metric.progressDirection.horizontal', {
+              defaultMessage: 'Horizontal',
+            }),
+            'data-test-subj': 'lnsMetric_progress_bar_horizontal',
+          },
+        ]}
+        idSelected={`${idPrefix}${state.progressDirection ?? 'vertical'}`}
+        onChange={(id) => {
+          const newDirection = id.replace(idPrefix, '') as LayoutDirection;
+          setState({
+            ...state,
+            progressDirection: newDirection,
+          });
+        }}
+      />
+    </EuiFormRow>
+  );
+}
+
+function PrimaryMetricEditor(props: Props) {
+  const { state, setState, frame, accessor } = props;
+
+  const [isPaletteOpen, setIsPaletteOpen] = useState(false);
+
+  const currentData = frame.activeData?.[state.layerId];
+
+  if (accessor == null || !isNumericFieldForDatatable(currentData, accessor)) {
+    return null;
+  }
+
+  const hasDynamicColoring = Boolean(state?.palette);
+
+  const startWithPercentPalette = Boolean(state.maxAccessor || state.breakdownByAccessor);
+
+  const activePalette = state?.palette || {
+    type: 'palette',
+    name: (startWithPercentPalette ? defaultPercentagePaletteParams : defaultNumberPaletteParams)
+      .name,
+    params: {
+      ...(startWithPercentPalette ? defaultPercentagePaletteParams : defaultNumberPaletteParams),
+    },
+  };
+
+  const currentMinMax = getDataBoundsForPalette(
+    {
+      metric: state.metricAccessor!,
+      max: state.maxAccessor,
+      breakdownBy: state.breakdownByAccessor,
+    },
+    frame.activeData?.[state.layerId]
+  );
+
+  const displayStops = applyPaletteParams(props.paletteService, activePalette, {
+    min: currentMinMax.min ?? DEFAULT_MIN_STOP,
+    max: currentMinMax.max ?? DEFAULT_MAX_STOP,
+  });
+
+  const togglePalette = () => setIsPaletteOpen(!isPaletteOpen);
+
+  const idPrefix = htmlIdGenerator()();
```

Comment:
nit: moving it to outside scope of the component makes the id more stable, otherwise the id gets changed with every rerender - I didn't notice any behaviour it would impact, but it's probably not a good practice, especially because the name stays the same: 

https://user-images.githubusercontent.com/4283304/181040231-bbe97c82-d5f2-47b3-b08d-f6869fe33a81.mov

Reply:
Can definitely move the ID up the scope, but it seems like the name staying the same is an EUI bug?